EarlGrey/Core/GREYKeyboard.m

```diff
@@ -36,9 +37,9 @@
 static GREYTapAction *gTapKeyAction;
 
 /**
- *  Flag set to @c YES when the keyboard is shown, @c NO when keyboard is hidden.
+ *  Flag set to @c true when the keyboard is shown, @c false when keyboard is hidden.
  */
-static BOOL gIsKeyboardShown = NO;
+static atomic_bool gIsKeyboardShown = false;
```

Comment:
Since this just a value var, does using volatile work? You can declare this as volatile and then use the variable normally:
static volatile BOOL gIsKeyboardShown = NO;